pkg/ptree/scanner.go

```
@@ -1,5 +1,132 @@
 package ptree
 
+import (
+	"bufio"
+	"io/ioutil"
+	"k8s.io/klog"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+)
+const(
+	QOSGuaranteed = "guaranteed"
+	QOSBurstable  = "burstable"
+	QOSBestEffort = "bestEffort"
+	CgroupBase    = "/sys/fs/cgroup/memory"
+	PodPrefix     = "pod"
+	CGROUP_PROCS = "cgroup.procs"
+)
+
+var (
+	kubeRoot = []string{"kubepods"}
+)
+
 type Scanner interface {
-	Scan(UID, QOS string) (Pod, error)
+	Scan(UID, QOS string) (*Pod, error)
+}
+
+type ScannerImpl struct{
+	pod       *Pod
+	container *Container
+	process   *Process
+	node      *Node
+}
+
+type CgroupName []string
+
+func (scan *ScannerImpl) Scan(UID, QOS string) (*Pod, error){
+	return scan.getContainers(NewPod(UID, QOS)), nil
```

Should return the right error here

---

ok